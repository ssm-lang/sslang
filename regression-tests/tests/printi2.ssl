putip_ putc x =
  if x < 10
    putc (x + 48)
  else
    putip_ putc (x / 10)
    putc (48 + x % 10)

puti_ putc x =
  if x < 0
    putc 45 // -
    putip_ putc (0 - x)
  else
    putip_ putc x

main cin cout =
  let putc c =
       after 1, cout <- c
       wait cout
  let puti = puti_ putc
  puti 123456789
  putc 10
  puti 0
  putc 10
  puti 1
  putc 10
  puti 9
  putc 10
  puti 10
  putc 10
  puti 12
  putc 10
  puti 19
  putc 10
  puti 99
  putc 10
  puti 100
  putc 10
  puti 901
  putc 10
  puti (0-1)
  putc 10
  puti (0-19)
  putc 10
  puti (0-987654321)
  putc 10