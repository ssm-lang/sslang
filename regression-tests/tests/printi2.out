123456789
0
1
9
10
12
19
99
100
901
-1
-19
-987654321
